webapp/components/integrations/components/confirm_integration.jsx

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+
+import BackstageHeader from 'components/backstage/components/backstage_header.jsx';
+import {FormattedMessage, FormattedHTMLMessage} from 'react-intl';
+import {browserHistory, Link} from 'react-router/es6';
+import SpinnerButton from 'components/spinner_button.jsx';
+
+import UserStore from 'stores/user_store.jsx';
+import IntegrationStore from 'stores/integration_store.jsx';
+
+import Constants from 'utils/constants.jsx';
+
+export default class ConfirmIntegration extends React.Component {
+    static get propTypes() {
+        return {
+            team: React.propTypes.object.isRequired,
+            location: React.PropTypes.object
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.handleDone = this.handleDone.bind(this);
+
+        this.handleIntegrationChange = this.handleIntegrationChange.bind(this);
+
+        const userId = UserStore.getCurrentId();
+
+        this.state = {
+            type: '',
+            id: '',
+            oauthApps: IntegrationStore.getOAuthApps(userId),
+            loading: !IntegrationStore.hasReceivedOAuthApps(userId)
+        };
+    }
+
+    componentWillMount() {
```

Comment:
Unless there's a specific reason you used `componentWillMount` then it shouldn't be used. All the state setting here should happen in the constructor and the listeners should be added in `componentDidMount`. `componentWillMount` has been "soft" deprecated and was essentially a way to have constructors for components before ES6